src/main/java/com/github/jonatino/process/DataSource.java

```diff
@@ -27,6 +27,8 @@
 public interface DataSource {
 
 	MemoryBuffer read(Pointer address, int size);
+	
+	MemoryBuffer read(Pointer address, int size, MemoryBuffer tobuf);
```

Comment:
I'd rather have `MemoryBuffer read(Pointer address, int size, MemoryBuffer tobuf)` to load data to other MemoryBuffers, so I can easily map and access the copied memory.

I use **Ubuntu MATE 16.04** with OpenJDK 1.8.0b121. JProfiler helped me found MASSIVE memory leaks and really inefficient use of memory. Garbage collector would pause my program every now and then cleaning millions of useless variables. I already solved it in [my repo](https://github.com/ericek111/Java-Memory-Manipulation/) by using my own wrapper for process_vm_readv/process_vm_writev.